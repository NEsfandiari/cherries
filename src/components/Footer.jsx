import React from 'react'
import Styled from 'styled-components'
import { FooterSubscription, FooterLinks } from './atoms'
import { StaticQuery, graphql } from 'gatsby'

const Container = Styled.div`
  position: relative;
  background-color: #E20031;
  font-family: Montserrat;
  font-size: .8rem;
  font-weight: 700;
  color: white;
  height: 15vw;
  width: 100vw;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  z-index: 2;
`

export default () => (
  <StaticQuery
    query={graphql`
      {
        allContentfulFooter {
          edges {
            node {
              footerSubscriptionTitle
              footerSubscriptionText
              footerLinks {
                content {
                  content {
                    data {
                      uri
                    }
                    content {
                      value
                    }
                  }
                }
              }
            }
          }
        }
      }
    `}
    render={data => (
      <Container>
        <FooterSubscription
          title={data.allContentfulFooter.edges[0].node.footerSubscriptionTitle}
          text={data.allContentfulFooter.edges[0].node.footerSubscriptionText}
        />
        <FooterLinks
          linkData={data.allContentfulFooter.edges[0].node.footerLinks.content[0].content.map(
            c => c[1]
          )}
        />
      </Container>
    )}
  />
)
