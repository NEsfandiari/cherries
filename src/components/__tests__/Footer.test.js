import React from 'react'
import renderer from 'react-test-renderer'
import { Footer, PureFooter } from '../Footer'

describe('Footer', () => {
  it('renders correctly', () => {
    
    const data = {
      allContentfulFooter: {
        edges: [
          {
            node: {
              footerSubscriptionTitle: 'Footer',
              footerSubscriptionText: 'Enter Footer text here, boiiiiii',
              footerLinks: {
                content: [
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://lipslut.com',
                        },
                        content: [
                          {
                            value: 'Refer a Friend',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://www.cherries.com/signin',
                        },
                        content: [
                          {
                            value: 'My Account',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://www.lipslut.com/contact-us/',
                        },
                        content: [
                          {
                            value: 'Careers',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://www.lipslut.com/faq/',
                        },
                        content: [
                          {
                            value: 'Help',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://www.lipslut.com/about-us/',
                        },
                        content: [
                          {
                            value: 'Our Story',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://www.instagram.com/hello.lipslut/',
                        },
                        content: [
                          {
                            value: 'Instagram',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://www.lipslut.com/terms-conditions/',
                        },
                        content: [
                          {
                            value: 'Terms of User',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                  {
                    content: [
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                      {
                        data: {
                          uri: 'https://www.lipslut.com/privacy-policy/',
                        },
                        content: [
                          {
                            value: 'Privacy',
                          },
                        ],
                      },
                      {
                        data: {
                          uri: null,
                        },
                        content: null,
                      },
                    ],
                  },
                ],
              },
            },
          },
        ],
      },
    }
    const tree = renderer.create(<PureFooter data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
