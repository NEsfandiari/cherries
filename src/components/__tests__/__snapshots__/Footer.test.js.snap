// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="sc-jTzLTM ehvlHX"
>
  <div
    className="sc-gqjmRU cOPKih"
  >
    <div>
      <h4>
        Footer
      </h4>
      <p
        style={
          Object {
            "maxWidth": "20.8rem",
          }
        }
      >
        Enter Footer text here, boiiiiii
      </p>
      <form>
        <label
          htmlFor="email"
        >
          Email
        </label>
        <br />
        <input
          style={
            Object {
              "width": "16rem",
            }
          }
          type="email"
        />
        <button
          style={
            Object {
              "backgroundColor": "#13CE66",
              "border": "none",
              "color": "#47525E",
              "height": "32px",
              "marginLeft": "12px",
              "paddingTop": "2px",
              "width": "6.5rem",
            }
          }
        >
          Sign Up
        </button>
      </form>
    </div>
  </div>
  <div
    className="sc-VigVT jYNWCi"
  >
    <ul
      className="list"
    >
      <li>
        <a
          className="list-items"
          href="https://lipslut.com"
        >
          Refer a Friend
        </a>
      </li>
      <li>
        <a
          className="list-items"
          href="https://www.cherries.com/signin"
        >
          My Account
        </a>
      </li>
      <li>
        <a
          className="list-items"
          href="https://www.lipslut.com/contact-us/"
        >
          Careers
        </a>
      </li>
      <li>
        <a
          className="list-items"
          href="https://www.lipslut.com/faq/"
        >
          Help
        </a>
      </li>
      <li>
        <a
          className="list-items"
          href="https://www.lipslut.com/about-us/"
        >
          Our Story
        </a>
      </li>
      <li>
        <a
          className="list-items"
          href="https://www.instagram.com/hello.lipslut/"
        >
          Instagram
        </a>
      </li>
      <li>
        <a
          className="list-items"
          href="https://www.lipslut.com/terms-conditions/"
        >
          Terms of User
        </a>
      </li>
      <li>
        <a
          className="list-items"
          href="https://www.lipslut.com/privacy-policy/"
        >
          Privacy
        </a>
      </li>
    </ul>
  </div>
</div>
`;
